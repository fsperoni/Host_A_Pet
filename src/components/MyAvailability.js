import React, { useContext, useState } from "react";
import UserContext from "../hooks/useUserContext";
import HostAPetApi from "./Api";
import AvailabilityEditForm from "./AvailabilityEditForm";

const MyAvailability = ({ avail, avails, setAvails, roles }) => {
  const [showEditForm, setShowEditForm] = useState(false);
  const { currentUser } = useContext(UserContext);

  const handleEditAvail = () => {
    setShowEditForm(!showEditForm);
  }

  const handleDeleteAvail = async (evt) => {
    const id = parseInt(evt.target.children[0].value);
    await HostAPetApi.deleteAvailability(currentUser.username, id);
    setAvails(avails => avails.filter(avail => avail.id !== id));
  }
  const role = roles.find(role => role.id === avail.roleId);
  

  return (
    <tr key={avail.id} >
      <td>{role.name}</td>
      <td>{avail.startDate.substring(0,10)}</td>
      <td>{avail.endDate.substring(0,10)}</td>
      <td>
        <div className="container">
          <button className="btn btn-sm btn-primary" onClick={handleEditAvail}>
            Edit
          </button>
          <button className="btn btn-sm btn-danger ms-2" onClick={handleDeleteAvail}>
            <data value={avail.id} />
            Delete
          </button>
          {showEditForm ?
            <AvailabilityEditForm avail={avail} avails={avails} setAvails={setAvails}
              setShowEditForm={setShowEditForm} roles={roles} /> :
            null}
        </div>
      </td>
    </tr>
  )
}

export default MyAvailability;