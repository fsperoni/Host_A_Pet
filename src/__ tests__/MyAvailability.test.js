import { render } from "@testing-library/react";
import MyAvailability from "../components/MyAvailability";
import UserContext from "../hooks/useUserContext";

const {
  currentAdminUser,
  avail, avails, roles
} = require("./_testCommon");

// smoke test
it('renders without crashing', function() {
  render(
    <UserContext.Provider
          value={{ currentUser: currentAdminUser() , function() {}}}>
      <MyAvailability setAvails={function() {}} avail={avail()} 
        avails={avails()} roles={roles()}/>
    </UserContext.Provider>);
});

//snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(
    <UserContext.Provider
          value={{ currentUser: currentAdminUser() , function() {}}}>
      <MyAvailability setAvails={function() {}} avail={avail()} 
        avails={avails()} roles={roles()}/>
    </UserContext.Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("should have 'Delete' and 'Edit' in the component.", function() {
  const { getByText } = render(
    <UserContext.Provider
          value={{ currentUser: currentAdminUser() , function() {}}}>
      <MyAvailability setAvails={function() {}} avail={avail()} 
        avails={avails()} roles={roles()}/>
    </UserContext.Provider>
  );
  expect(getByText("Delete")).toBeInTheDocument();
  expect(getByText("Edit")).toBeInTheDocument();
});
