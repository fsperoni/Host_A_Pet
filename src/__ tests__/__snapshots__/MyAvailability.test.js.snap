// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  Host2022-04-022022-04-20
  <div
    class="container"
  >
    <button
      class="btn btn-sm btn-primary"
    >
      Edit
    </button>
    <button
      class="btn btn-sm btn-danger ms-2"
    >
      <data
        value="1"
      />
      Delete
    </button>
  </div>
</DocumentFragment>
`;
